tool/tsh/proxy.go

```diff
@@ -197,40 +204,52 @@ func onProxyCommandDB(cf *CLIConf) error {
 	return nil
 }
 
-func mkLocalProxy(cf *CLIConf, remoteProxyAddr string, protocol string, listener net.Listener) (*alpnproxy.LocalProxy, error) {
-	alpnProtocol, err := toALPNProtocol(protocol)
+type localProxyOpts struct {
+	proxyAddr string
+	listener  net.Listener
+	protocol  string
+	insecure  bool
+	certFile  string
+	keyFile   string
+}
+
+func mkLocalProxy(ctx context.Context, opts localProxyOpts) (*alpnproxy.LocalProxy, error) {
+	alpnProtocol, err := alpncommon.ToALPNProtocol(opts.protocol)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	address, err := utils.ParseAddr(opts.proxyAddr)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	address, err := utils.ParseAddr(remoteProxyAddr)
+	certs, err := mkLocalProxyCerts(opts.certFile, opts.keyFile)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 	lp, err := alpnproxy.NewLocalProxy(alpnproxy.LocalProxyConfig{
-		InsecureSkipVerify: cf.InsecureSkipVerify,
-		RemoteProxyAddr:    remoteProxyAddr,
+		InsecureSkipVerify: opts.insecure,
+		RemoteProxyAddr:    opts.proxyAddr,
 		Protocol:           alpnProtocol,
-		Listener:           listener,
-		ParentContext:      cf.Context,
+		Listener:           opts.listener,
+		ParentContext:      ctx,
 		SNI:                address.Host(),
+		Certs:              certs,
 	})
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 	return lp, nil
 }
 
-func toALPNProtocol(dbProtocol string) (alpncommon.Protocol, error) {
-	switch dbProtocol {
-	case defaults.ProtocolMySQL:
-		return alpncommon.ProtocolMySQL, nil
-	case defaults.ProtocolPostgres, defaults.ProtocolCockroachDB:
-		return alpncommon.ProtocolPostgres, nil
-	case defaults.ProtocolMongoDB:
-		return alpncommon.ProtocolMongoDB, nil
-	default:
-		return "", trace.NotImplemented("%q protocol is not supported", dbProtocol)
+func mkLocalProxyCerts(certFile, keyFile string) ([]tls.Certificate, error) {
+	if certFile == "" || keyFile == "" {
```

Comment:
Can we return an error if only `certFile` or `keyFIle` is provided? I can image someone passing only one of them and not knowing why connection is failing.